packages/utils/src/retry.ts

```diff
@@ -36,7 +37,7 @@ export async function retry<A>(fn: (attempt: number) => A | Promise<A>, opts?: R
       if (shouldRetry && !shouldRetry(lastError)) {
         break;
       } else if (opts?.retryDelay !== undefined) {
-        await sleep(opts?.retryDelay);
+        await sleep(opts?.retryDelay, opts?.signal);
```

Comment:
This change allows to clear timers for long running retries methods in case of node shutdown.